File: MPI/MPI_day1_2022/ranksize.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv)
{
    int size, rank;
    MPI_Init(&argc, &argv);
    MPI_Comm_size(MPI_COMM_WORLD, &size);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    printf("I am rank %d out of %d processors\n",rank,size);
    MPI_Finalize();
    return 0;
}

File: MPI/MPI_day1_2022/ranksize.f
      program main
      include 'mpif.h'

      integer ierr, rank, size

      call MPI_INIT(ierr)
      call MPI_COMM_SIZE(MPI_COMM_WORLD,size,ierr)
      call MPI_COMM_RANK(MPI_COMM_WORLD,rank,ierr)
      print*, "I am rank", rank, " out of ", size, "processors"
      call MPI_FINALIZE(ierr)
      end
